### PR TITLE
Update eslint and eslintrc

### LINKED ISSUE #32: Update eslint
Update eslint, adjust its configuration and fix issues caused by this.

### PR DESCRIPTION
**Pull request by lxanders**
Fixes #32

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -28,13 +38,13 @@
         "no-extend-native": 2,
         "no-extra-bind": 2,
         "no-extra-boolean-cast": 2,
-        "no-extra-parens": 2,
+        "no-extra-parens": 0,
```
Comment (lo1tuma) [inline .eslintrc]: Why was this disabled?

File: .eslintrc [hunk 2]
```diff
@@ -67,7 +78,7 @@
         "no-proto": 2,
         "no-redeclare": 2,
         "no-regex-spaces": 2,
-        "no-reserved-keys": 2,
+        "no-reserved-keys": 0,
```
Comment (lo1tuma) [inline .eslintrc]: Why was this disabled?

File: .eslintrc [hunk 3]
```diff
@@ -75,28 +86,40 @@
...
         "no-sync": 2,
         "no-ternary": 0,
         "no-trailing-spaces": 2,
+        "no-throw-literal": 2,
         "no-undef": 2,
         "no-undef-init": 2,
-        "no-undefined": 2,
-        "no-underscore-dangle": 2,
+        "no-undefined": 0,
+        "no-underscore-dangle": 0,
```
Comment (lo1tuma) [inline .eslintrc]: I'm fine with disabling `no-undefined` but why was `no-underscore-dangle` disabled?

**Comment by lxanders** [comment]
Thanks for the review! You are right, these rules might not need to be activated in this project. I will have a look and rework this.

**Pull request merged**